# ITMGT 25.03 EE
# Assignment 3A

This is a data-cleaning assignment and is the first of two parts. The second part (3B) will be given Monday or Tuesday next week. This notebook is uploaded to our Github repository along with a csv of fake, dirty data called **recweekdata.csv**.


The output of this assignment is to convert recweekdata.csv into processable data for statistics. Submit the following files through GitHub:
1. A cleaned recweekdata.csv called cleanrecweekdata.csv **(5 points)**
2. This notebook, containing the code (properly commented) that you used to clean the data **(5 points)**
3. An html file of summary statistics from each column called **stats.html**. The requirements for the summary statistics are described below. **(10 points)**

This entire assignment (3A) is worth 20 points.

Upload your output files to your Github repositories. Make sure to send an email to **joben.ilagan@gmail.com** once you're done.

This assignment is due on **Tuesday, July 9, 2019** at 12:00NN.

In [12]:
## Code goes here


In [13]:
import csv
with open('recweekdata.csv') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    line_count = 0
    for row in csv_reader:
        if line_count == 0:
            print(f'Column names are {", ".join(row)}')
        else:
            print(f'\t{row[0]}  {row[1]}  {row[2]} {row[3]}  {row[4]}  {row[5]} {row[6]}')
        line_count += 1

Column names are firstname, lastname, birthmonth, birthday, birthyear, course, phonenumber
	name1  lastname1  11 26  1998  BS LM +63 918 302 6210
	name2  lastname2  May 26  2001  Management 927 212 6276
	name3  lastname3  5 16  1997  BS CTM +63 918 253 7017
	name4  lastname4  Dec 9  97  BS ITE 918 714 4360
	name5  lastname5  Sep 5  97  BS CTM +63 927 462 1674
	name6  lastname6  March 2  1999  Management +63 918 629 5268
	name7  lastname7  11 10  98  BS LM +63 917 369 7388
	name8  lastname8  Aug 28  99  Management Engineering 917 240 6000
	name9  lastname9  July 10  97  Management Engineering 918 226 5748
	name10  lastname10  9 14  2001  CTM +63 918 342 1717
	name11  lastname11  December 8  98  BS LM +63 927 708 1308
	name12  lastname12  5 4  97  BS ME 927 747 5497
	name13  lastname13  January 28  1999  BS CTM +63 917 959 4490
	name14  lastname14  1 11  1999  BS MGT +63 927 543 4192
	name15  lastname15  6 6  98  Information Technology Entrepreneurship 917 211 9497
	name16  lastname16  

	name1724  lastname1724  Jun 10  1997  MGT 917 252 6526
	name1725  lastname1725  April 23  1998  CTM +63 927 268 5622
	name1726  lastname1726  September 26  97  BS LM +63 917 660 9137
	name1727  lastname1727  1 26  98  Management Engineering +63 927 581 4936
	name1728  lastname1728  8 19  1997  Management +63 917 565 5934
	name1729  lastname1729  Nov 12  1997  BS LM 918 776 4007
	name1730  lastname1730  April 13  1999  ME +63 918 461 2548
	name1731  lastname1731  8 22  1998  BS CTM +63 927 368 7728
	name1732  lastname1732  Aug 21  98  Management 927 376 2938
	name1733  lastname1733  Jan 19  2000  Communications Technology +63 927 740 2743
	name1734  lastname1734  Jun 15  2000  BS LM 918 200 7458
	name1735  lastname1735  3 26  1997  BS LM +63 927 632 8106
	name1736  lastname1736  Oct 17  97  LM 917 653 7150
	name1737  lastname1737  Aug 1  2001  MGT +63 917 932 2383
	name1738  lastname1738  Feb 4  2001  LM 927 173 7762
	name1739  lastname1739  8 26  2000  MGT 918 647 9671
	name1740  last

In [158]:
firstname_list = list() # clean list of first names

with open('recweekdata.csv') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    line_count = 0
    for row in csv_reader:
        if line_count == 0:
            pass
        else:
            firstname_list.append(row[0])
        line_count += 1
        
print(firstname_list)

['name1', 'name2', 'name3', 'name4', 'name5', 'name6', 'name7', 'name8', 'name9', 'name10', 'name11', 'name12', 'name13', 'name14', 'name15', 'name16', 'name17', 'name18', 'name19', 'name20', 'name21', 'name22', 'name23', 'name24', 'name25', 'name26', 'name27', 'name28', 'name29', 'name30', 'name31', 'name32', 'name33', 'name34', 'name35', 'name36', 'name37', 'name38', 'name39', 'name40', 'name41', 'name42', 'name43', 'name44', 'name45', 'name46', 'name47', 'name48', 'name49', 'name50', 'name51', 'name52', 'name53', 'name54', 'name55', 'name56', 'name57', 'name58', 'name59', 'name60', 'name61', 'name62', 'name63', 'name64', 'name65', 'name66', 'name67', 'name68', 'name69', 'name70', 'name71', 'name72', 'name73', 'name74', 'name75', 'name76', 'name77', 'name78', 'name79', 'name80', 'name81', 'name82', 'name83', 'name84', 'name85', 'name86', 'name87', 'name88', 'name89', 'name90', 'name91', 'name92', 'name93', 'name94', 'name95', 'name96', 'name97', 'name98', 'name99', 'name100', 'name10

In [15]:
lastname_list = list() # clean list of last names

with open('recweekdata.csv') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    line_count = 0
    for row in csv_reader:
        if line_count == 0:
            pass
        else:
            lastname_list.append(row[1])
        line_count += 1
        
lastname_list

['lastname1',
 'lastname2',
 'lastname3',
 'lastname4',
 'lastname5',
 'lastname6',
 'lastname7',
 'lastname8',
 'lastname9',
 'lastname10',
 'lastname11',
 'lastname12',
 'lastname13',
 'lastname14',
 'lastname15',
 'lastname16',
 'lastname17',
 'lastname18',
 'lastname19',
 'lastname20',
 'lastname21',
 'lastname22',
 'lastname23',
 'lastname24',
 'lastname25',
 'lastname26',
 'lastname27',
 'lastname28',
 'lastname29',
 'lastname30',
 'lastname31',
 'lastname32',
 'lastname33',
 'lastname34',
 'lastname35',
 'lastname36',
 'lastname37',
 'lastname38',
 'lastname39',
 'lastname40',
 'lastname41',
 'lastname42',
 'lastname43',
 'lastname44',
 'lastname45',
 'lastname46',
 'lastname47',
 'lastname48',
 'lastname49',
 'lastname50',
 'lastname51',
 'lastname52',
 'lastname53',
 'lastname54',
 'lastname55',
 'lastname56',
 'lastname57',
 'lastname58',
 'lastname59',
 'lastname60',
 'lastname61',
 'lastname62',
 'lastname63',
 'lastname64',
 'lastname65',
 'lastname66',
 'lastname67',
 'la

In [16]:
dirty_course_list = list()

with open('recweekdata.csv') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    line_count = 0
    for row in csv_reader:
        if line_count == 0:
            pass
        else:
            dirty_course_list.append(row[5])
        line_count += 1
        
dirty_course_list

['BS LM',
 'Management',
 'BS CTM',
 'BS ITE',
 'BS CTM',
 'Management',
 'BS LM',
 'Management Engineering',
 'Management Engineering',
 'CTM',
 'BS LM',
 'BS ME',
 'BS CTM',
 'BS MGT',
 'Information Technology Entrepreneurship',
 'BS LM',
 'Information Technology Entrepreneurship',
 'BS MGT',
 'Information Technology Entrepreneurship',
 'BS ME',
 'LM',
 'ITE',
 'MGT',
 'ME',
 'BS ME',
 'BS ME',
 'CTM',
 'BS LM',
 'LM',
 'Communications Technology',
 'BS ITE',
 'Management',
 'BS ITE',
 'BS LM',
 'ITE',
 'BS MGT',
 'MGT',
 'Legal Management',
 'ME',
 'BS LM',
 'BS MGT',
 'BS LM',
 'Management',
 'Information Technology Entrepreneurship',
 'Management Engineering',
 'ME',
 'Management',
 'Information Technology Entrepreneurship',
 'Legal Management',
 'BS LM',
 'ITE',
 'Communications Technology',
 'BS MGT',
 'BS CTM',
 'Management Engineering',
 'ME',
 'BS CTM',
 'BS ME',
 'Information Technology Entrepreneurship',
 'ITE',
 'Communications Technology',
 'ITE',
 'BS ME',
 'Information 

In [17]:
unique_names = set(dirty_course_list) # removing repeated names

unique_names

{'BS CTM',
 'BS ITE',
 'BS LM',
 'BS ME',
 'BS MGT',
 'CTM',
 'Communications Technology',
 'ITE',
 'Information Technology Entrepreneurship',
 'LM',
 'Legal Management',
 'ME',
 'MGT',
 'Management',
 'Management Engineering'}

In [18]:
[n for n in unique_names] # making it a list

['ME',
 'BS LM',
 'Communications Technology',
 'Management Engineering',
 'Information Technology Entrepreneurship',
 'BS ME',
 'BS MGT',
 'BS CTM',
 'CTM',
 'BS ITE',
 'Legal Management',
 'MGT',
 'Management',
 'LM',
 'ITE']

In [19]:
unique_names_dictionary = {name:None for name in unique_names}

unique_names_dictionary

{'ME': None,
 'BS LM': None,
 'Communications Technology': None,
 'Management Engineering': None,
 'Information Technology Entrepreneurship': None,
 'BS ME': None,
 'BS MGT': None,
 'BS CTM': None,
 'CTM': None,
 'BS ITE': None,
 'Legal Management': None,
 'MGT': None,
 'Management': None,
 'LM': None,
 'ITE': None}

In [21]:
for i in unique_names_dictionary:
    final_name = input(i+": "+"Enter name: ")
    unique_names_dictionary[i]=final_name

ME: Enter name: BS Management Engineering
BS LM: Enter name: BS Legal Management
Communications Technology: Enter name: BS Communications Technology
Management Engineering: Enter name: BS Management Engineering
Information Technology Entrepreneurship: Enter name: BS Information Technology Entrepreneurship
BS ME: Enter name: BS Management Engineering
BS MGT: Enter name: BS Management
BS CTM: Enter name: BS Communications Technology
CTM: Enter name: BS Communications Technology
BS ITE: Enter name: BS Information Technology Entrepreneurship
Legal Management: Enter name: BS Legal Management
MGT: Enter name: BS Management
Management: Enter name: BS Management
LM: Enter name: BS Legal Management
ITE: Enter name: BS Information Technology Entrepreneurship


In [22]:
unique_names_dictionary

{'ME': 'BS Management Engineering',
 'BS LM': 'BS Legal Management',
 'Communications Technology': 'BS Communications Technology',
 'Management Engineering': 'BS Management Engineering',
 'Information Technology Entrepreneurship': 'BS Information Technology Entrepreneurship',
 'BS ME': 'BS Management Engineering',
 'BS MGT': 'BS Management',
 'BS CTM': 'BS Communications Technology',
 'CTM': 'BS Communications Technology',
 'BS ITE': 'BS Information Technology Entrepreneurship',
 'Legal Management': 'BS Legal Management',
 'MGT': 'BS Management',
 'Management': 'BS Management',
 'LM': 'BS Legal Management',
 'ITE': 'BS Information Technology Entrepreneurship'}

In [23]:
import re

In [24]:
clean_course_list = [re.sub(r''.join(course),r''.join(unique_names_dictionary[course]),course) for course in dirty_course_list]

clean_course_list

['BS Legal Management',
 'BS Management',
 'BS Communications Technology',
 'BS Information Technology Entrepreneurship',
 'BS Communications Technology',
 'BS Management',
 'BS Legal Management',
 'BS Management Engineering',
 'BS Management Engineering',
 'BS Communications Technology',
 'BS Legal Management',
 'BS Management Engineering',
 'BS Communications Technology',
 'BS Management',
 'BS Information Technology Entrepreneurship',
 'BS Legal Management',
 'BS Information Technology Entrepreneurship',
 'BS Management',
 'BS Information Technology Entrepreneurship',
 'BS Management Engineering',
 'BS Legal Management',
 'BS Information Technology Entrepreneurship',
 'BS Management',
 'BS Management Engineering',
 'BS Management Engineering',
 'BS Management Engineering',
 'BS Communications Technology',
 'BS Legal Management',
 'BS Legal Management',
 'BS Communications Technology',
 'BS Information Technology Entrepreneurship',
 'BS Management',
 'BS Information Technology Entrep

In [25]:
dirty_birthmonth_list = list()

with open('recweekdata.csv') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    line_count = 0
    for row in csv_reader:
        if line_count == 0:
            pass
        else:
            dirty_birthmonth_list.append(row[2])
        line_count += 1
        
# dirty_birthmonth_list

unique_birthmonths = set(dirty_birthmonth_list)

# unique_birthmonths

[n for n in unique_birthmonths]

unique_birthmonths_dictionary = {name:None for name in unique_birthmonths}

unique_birthmonths_dictionary

{'Aug': None,
 'Oct': None,
 'November': None,
 '9': None,
 'Jul': None,
 '8': None,
 'Nov': None,
 'Feb': None,
 'July': None,
 'February': None,
 'Dec': None,
 'Apr': None,
 'December': None,
 '12': None,
 'Jan': None,
 'Jun': None,
 '10': None,
 'May': None,
 '7': None,
 '4': None,
 'Sep': None,
 'January': None,
 'Mar': None,
 'June': None,
 '11': None,
 '1': None,
 'April': None,
 '2': None,
 '6': None,
 '3': None,
 'August': None,
 '5': None,
 'March': None,
 'September': None}

In [27]:
for i in unique_birthmonths_dictionary:
    final_name = input(i+": "+"Enter name: ")
    unique_birthmonths_dictionary[i]=final_name

Aug: Enter name: August
Oct: Enter name: October
November: Enter name: November
9: Enter name: September
Jul: Enter name: July
8: Enter name: August
Nov: Enter name: November
Feb: Enter name: February
July: Enter name: July
February: Enter name: February
Dec: Enter name: December
Apr: Enter name: April
December: Enter name: December
12: Enter name: December
Jan: Enter name: January
Jun: Enter name: June
10: Enter name: October
May: Enter name: May
7: Enter name: July
4: Enter name: April
Sep: Enter name: September
January: Enter name: January
Mar: Enter name: March
June: Enter name: June
11: Enter name: November
1: Enter name: January
April: Enter name: April
2: Enter name: February
6: Enter name: June
3: Enter name: March
August: Enter name: August
5: Enter name: May
March: Enter name: March
September: Enter name: September


In [28]:
unique_birthmonths_dictionary

{'Aug': 'August',
 'Oct': 'October',
 'November': 'November',
 '9': 'September',
 'Jul': 'July',
 '8': 'August',
 'Nov': 'November',
 'Feb': 'February',
 'July': 'July',
 'February': 'February',
 'Dec': 'December',
 'Apr': 'April',
 'December': 'December',
 '12': 'December',
 'Jan': 'January',
 'Jun': 'June',
 '10': 'October',
 'May': 'May',
 '7': 'July',
 '4': 'April',
 'Sep': 'September',
 'January': 'January',
 'Mar': 'March',
 'June': 'June',
 '11': 'November',
 '1': 'January',
 'April': 'April',
 '2': 'February',
 '6': 'June',
 '3': 'March',
 'August': 'August',
 '5': 'May',
 'March': 'March',
 'September': 'September'}

In [127]:
# clean birthmonths list

clean_birthmonths_list = [re.sub(r''.join(birthmonth),r''.join(unique_birthmonths_dictionary[birthmonth]),birthmonth) for birthmonth in dirty_birthmonth_list]

clean_birthmonths_list

['November',
 'May',
 'May',
 'December',
 'September',
 'March',
 'November',
 'August',
 'July',
 'September',
 'December',
 'May',
 'January',
 'January',
 'June',
 'April',
 'January',
 'April',
 'December',
 'March',
 'July',
 'September',
 'April',
 'October',
 'May',
 'December',
 'March',
 'November',
 'October',
 'May',
 'April',
 'July',
 'December',
 'October',
 'July',
 'December',
 'February',
 'August',
 'May',
 'February',
 'November',
 'September',
 'January',
 'February',
 'January',
 'December',
 'August',
 'April',
 'October',
 'March',
 'August',
 'March',
 'December',
 'July',
 'May',
 'September',
 'May',
 'April',
 'April',
 'August',
 'September',
 'December',
 'February',
 'November',
 'March',
 'September',
 'December',
 'November',
 'November',
 'May',
 'August',
 'December',
 'June',
 'October',
 'September',
 'February',
 'May',
 'September',
 'December',
 'August',
 'February',
 'November',
 'June',
 'February',
 'January',
 'November',
 'August',
 'Januar

In [151]:
sorted_birthmonths_list = []

for month in clean_birthmonths_list:
    if month == "January":
        sorted_birthmonths_list.append(month)
        continue
    else:
        pass

for month in clean_birthmonths_list:
    if month == "February":
        sorted_birthmonths_list.append(month)
        continue
    else:
        pass

for month in clean_birthmonths_list:
    if month == "March":
        sorted_birthmonths_list.append(month)
        continue
    else:
        pass
   
for month in clean_birthmonths_list:
    if month == "April":
        sorted_birthmonths_list.append(month)
        continue
    else:
        pass

for month in clean_birthmonths_list:
    if month == "May":
        sorted_birthmonths_list.append(month)
        continue
    else:
        pass

for month in clean_birthmonths_list:
    if month == "June":
        sorted_birthmonths_list.append(month)
        continue
    else:
        pass

for month in clean_birthmonths_list:
    if month == "July":
        sorted_birthmonths_list.append(month)
        continue
    else:
        pass

for month in clean_birthmonths_list:
    if month == "August":
        sorted_birthmonths_list.append(month)
        continue
    else:
        pass

for month in clean_birthmonths_list:
    if month == "September":
        sorted_birthmonths_list.append(month)
        continue
    else:
        pass
   
for month in clean_birthmonths_list:
    if month == "October":
        sorted_birthmonths_list.append(month)
        continue
    else:
        pass

for month in clean_birthmonths_list:
    if month == "November":
        sorted_birthmonths_list.append(month)
        continue
    else:
        pass

for month in clean_birthmonths_list:
    if month == "December":
        sorted_birthmonths_list.append(month)
        continue
    else:
        pass

sorted_birthmonths_list

['January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'January',
 'Ja

In [116]:
birthday_list = list() # making birthday a list

with open('recweekdata.csv') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    line_count = 0
    for row in csv_reader:
        if line_count == 0:
            pass
        else:
            birthday_list.append(row[3])
        line_count += 1
        
birthday_list

['26',
 '26',
 '16',
 '9',
 '5',
 '2',
 '10',
 '28',
 '10',
 '14',
 '8',
 '4',
 '28',
 '11',
 '6',
 '20',
 '3',
 '8',
 '28',
 '10',
 '13',
 '15',
 '15',
 '1',
 '6',
 '5',
 '4',
 '10',
 '10',
 '28',
 '6',
 '16',
 '17',
 '4',
 '14',
 '11',
 '2',
 '25',
 '3',
 '27',
 '16',
 '24',
 '13',
 '25',
 '27',
 '24',
 '23',
 '13',
 '11',
 '25',
 '22',
 '21',
 '15',
 '10',
 '8',
 '9',
 '9',
 '6',
 '24',
 '26',
 '8',
 '7',
 '22',
 '13',
 '23',
 '24',
 '9',
 '4',
 '19',
 '12',
 '19',
 '1',
 '21',
 '25',
 '17',
 '2',
 '19',
 '15',
 '23',
 '10',
 '1',
 '22',
 '18',
 '28',
 '10',
 '21',
 '15',
 '26',
 '26',
 '15',
 '16',
 '8',
 '14',
 '24',
 '6',
 '16',
 '19',
 '10',
 '12',
 '10',
 '2',
 '14',
 '16',
 '16',
 '5',
 '25',
 '21',
 '14',
 '15',
 '8',
 '22',
 '19',
 '27',
 '22',
 '26',
 '19',
 '11',
 '4',
 '27',
 '3',
 '5',
 '1',
 '23',
 '2',
 '24',
 '4',
 '3',
 '20',
 '25',
 '18',
 '23',
 '10',
 '18',
 '7',
 '3',
 '5',
 '5',
 '5',
 '8',
 '2',
 '8',
 '3',
 '15',
 '14',
 '9',
 '11',
 '6',
 '13',
 '18',
 '5',
 

In [121]:
birthday_list_new = [int(x) for x in birthday_list]
birthday_list_new.sort()

birthday_list_new

[1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,


In [119]:
birthday_list.sort()

birthday_list


['1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '1',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',
 '10',

In [31]:
dirty_birthyear_list = list()

with open('recweekdata.csv') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    line_count = 0
    for row in csv_reader:
        if line_count == 0:
            pass
        else:
            dirty_birthyear_list.append(row[4])
        line_count += 1
        
# dirty_birthyear_list

unique_birthyears = set(dirty_birthyear_list)

# unique_birthyears

[n for n in unique_birthyears]

unique_birthyears_dictionary = {name:None for name in unique_birthyears}

# unique_birthyears_dictionary

for i in unique_birthyears_dictionary:
    final_name = input(i+": "+"Enter name: ")
    unique_birthyears_dictionary[i]=final_name

2001: Enter name: 2001
97: Enter name: 1997
98: Enter name: 1998
1997: Enter name: 1997
2000: Enter name: 2000
1998: Enter name: 1998
99: Enter name: 1999
1999: Enter name: 1999


In [32]:
# unique_birthyears_dictionary

clean_birthyears_list = [re.sub(r''.join(birthyear),r''.join(unique_birthyears_dictionary[birthyear]),birthyear) for birthyear in dirty_birthyear_list]

clean_birthyears_list

['1998',
 '2001',
 '1997',
 '1997',
 '1997',
 '1999',
 '1998',
 '1999',
 '1997',
 '2001',
 '1998',
 '1997',
 '1999',
 '1999',
 '1998',
 '1999',
 '1997',
 '1997',
 '1997',
 '1997',
 '1999',
 '1999',
 '1997',
 '1997',
 '1997',
 '1999',
 '1997',
 '2000',
 '1999',
 '1999',
 '1999',
 '1999',
 '1999',
 '1999',
 '2001',
 '1998',
 '2000',
 '2001',
 '1997',
 '1998',
 '1999',
 '1998',
 '1999',
 '1998',
 '1997',
 '1998',
 '1999',
 '1999',
 '1997',
 '2001',
 '1998',
 '1999',
 '2000',
 '1998',
 '1998',
 '1998',
 '2001',
 '1997',
 '1998',
 '1999',
 '1998',
 '2000',
 '2000',
 '1997',
 '1997',
 '1997',
 '1997',
 '2001',
 '1998',
 '1998',
 '1999',
 '1999',
 '1998',
 '1997',
 '1998',
 '1997',
 '2000',
 '2000',
 '2000',
 '1999',
 '1997',
 '1997',
 '2000',
 '1999',
 '1999',
 '1998',
 '1998',
 '1997',
 '1998',
 '1999',
 '1997',
 '1997',
 '1998',
 '1997',
 '2001',
 '1999',
 '1997',
 '1998',
 '1998',
 '1999',
 '1997',
 '2001',
 '2000',
 '2001',
 '1999',
 '1997',
 '2001',
 '1999',
 '1998',
 '1999',
 '2001',
 

In [59]:
dirty_phonenumber_list = list()

with open('recweekdata.csv') as csv_file:
    csv_reader = csv.reader(csv_file, delimiter=',')
    line_count = 0
    for row in csv_reader:
        if line_count == 0:
            pass
        else:
            dirty_phonenumber_list.append(row[6])
        line_count += 1
        
# dirty_phonenumber_list

unique_phonenumbers = set(dirty_phonenumber_list)

# unique_phonenumbers

# [n for n in unique_phonenumbers]

# unique_phonenumbers_dictionary = {name:None for name in unique_phonenumbers}

# unique_phonenumbers_dictionary

phonenumbers_no_space = [n.replace(" ", "") for n in unique_phonenumbers] # removed spaces

# phonenumbers_no_space

# clean_phonenumber_list = list(a)

# clean_phonenumber_list

clean_phonenumber_list = []

for a in phonenumbers_no_space: # added +63 to the beginning of those without it
    if len(a) == 13:
        a = a
        clean_phonenumber_list.append(a)
    elif len(a) < 13:
        a = "+63" + a
        clean_phonenumber_list.append(a)
        
clean_phonenumber_list

['+639182677380',
 '+639188057805',
 '+639186815912',
 '+639271697384',
 '+639183516631',
 '+639271401589',
 '+639184011101',
 '+639278232501',
 '+639184018010',
 '+639171593829',
 '+639185764164',
 '+639173956151',
 '+639174169675',
 '+639177827480',
 '+639279161852',
 '+639176699647',
 '+639183165526',
 '+639179917278',
 '+639277393278',
 '+639271184679',
 '+639189094946',
 '+639187895546',
 '+639171108760',
 '+639186306211',
 '+639277666306',
 '+639178115416',
 '+639272342768',
 '+639188816528',
 '+639276444841',
 '+639271468146',
 '+639178522643',
 '+639272167464',
 '+639277035871',
 '+639178745948',
 '+639181532402',
 '+639173478315',
 '+639185369523',
 '+639276915341',
 '+639271817133',
 '+639182435342',
 '+639279231234',
 '+639278814083',
 '+639188727265',
 '+639184318667',
 '+639279975569',
 '+639186991503',
 '+639279408616',
 '+639173208047',
 '+639189806630',
 '+639176508776',
 '+639182093758',
 '+639185104996',
 '+639272659044',
 '+639279338376',
 '+639274922596',
 '+6391791

In [160]:
rows = zip(firstname_list, lastname_list, clean_birthmonths_list, birthday_list, clean_birthyears_list, clean_phonenumber_list)



In [161]:
import csv

with open('cleanrecweekdata', 'a', newline='') as csvfile:
    order_writer = csv.writer(csvfile, quoting=csv.QUOTE_MINIMAL)
    for row in rows:
        order_writer.writerow(row)

# Required Summary Statistics
1. How many people were born in each month?
2. How many people were born on each day?
3. How many people were born in each year?
4. How many people come from each course?
5. How many people have numbers from Globe (0917 or 0927) and Smart (0918)? *Note: it's up to you how to format this output*

Answer these questions in an html file called stats.html. The format of the html is as follows (in template form, so you can use this as format string in your program, though feel free to modify with variable names if you find it easier to do so):


```
<html>
    <body>
         <h2>Summary Statistics</h2>
         <table>
             <tr><th>Stat</th><th>Value</th></tr>
             <tr><td>People born in each month</td><td>{}</td></tr>
             <tr><td>People born on each day</td><td>{}</td></tr>
             <tr><td>People born in each year</td><td>{}</td></tr>
             <tr><td>People coming from each course</td><td>{}</td></tr>
             <tr><td>People having numbers from Globe and Smart</td><td>{}</td></tr>
         </table>
    </body>
</html>
```

In [152]:
mytally_birthmonths = {}

for word in sorted_birthmonths_list:
    mytally_birthmonths.setdefault(word,sorted_birthmonths_list.count(word))
    
mytally_birthmonths

{'January': 270,
 'February': 232,
 'March': 246,
 'April': 267,
 'May': 273,
 'June': 259,
 'July': 249,
 'August': 241,
 'September': 232,
 'October': 178,
 'November': 277,
 'December': 276}

In [122]:
mytally_birthdays = {}

for word in birthday_list_new:
    mytally_birthdays.setdefault(word,birthday_list_new.count(word))
    
mytally_birthdays

{1: 112,
 2: 106,
 3: 90,
 4: 110,
 5: 100,
 6: 99,
 7: 88,
 8: 111,
 9: 98,
 10: 111,
 11: 105,
 12: 98,
 13: 119,
 14: 102,
 15: 121,
 16: 115,
 17: 116,
 18: 107,
 19: 113,
 20: 91,
 21: 123,
 22: 103,
 23: 116,
 24: 108,
 25: 116,
 26: 116,
 27: 102,
 28: 104}

In [153]:
# unnecessary

# mytally_birthdays_sorted = sorted(mytally_birthdays.items())
# mytally_birthdays_sorted

In [154]:
mytally_birthyears = {}

for word in clean_birthyears_list:
    mytally_birthyears.setdefault(word,clean_birthyears_list.count(word))
    
mytally_birthyears

{'1998': 746, '2001': 362, '1997': 756, '1999': 756, '2000': 380}

In [155]:
mytally_birthyears_sorted = sorted(mytally_birthyears.items())
mytally_birthyears_sorted

[('1997', 756), ('1998', 746), ('1999', 756), ('2000', 380), ('2001', 362)]

In [156]:
mytally_courses = {}

for word in clean_course_list:
    mytally_courses.setdefault(word,clean_course_list.count(word))
    
mytally_courses

{'BS Legal Management': 585,
 'BS Management': 629,
 'BS Communications Technology': 589,
 'BS Information Technology Entrepreneurship': 590,
 'BS Management Engineering': 607}

In [162]:
beginning_phonenumbers_list = []

for n in clean_phonenumber_list:
    beginning_phonenumbers_list.append(n[0:6])

beginning_phonenumbers_list

['+63918',
 '+63918',
 '+63918',
 '+63927',
 '+63918',
 '+63927',
 '+63918',
 '+63927',
 '+63918',
 '+63917',
 '+63918',
 '+63917',
 '+63917',
 '+63917',
 '+63927',
 '+63917',
 '+63918',
 '+63917',
 '+63927',
 '+63927',
 '+63918',
 '+63918',
 '+63917',
 '+63918',
 '+63927',
 '+63917',
 '+63927',
 '+63918',
 '+63927',
 '+63927',
 '+63917',
 '+63927',
 '+63927',
 '+63917',
 '+63918',
 '+63917',
 '+63918',
 '+63927',
 '+63927',
 '+63918',
 '+63927',
 '+63927',
 '+63918',
 '+63918',
 '+63927',
 '+63918',
 '+63927',
 '+63917',
 '+63918',
 '+63917',
 '+63918',
 '+63918',
 '+63927',
 '+63927',
 '+63927',
 '+63917',
 '+63917',
 '+63917',
 '+63917',
 '+63918',
 '+63917',
 '+63917',
 '+63917',
 '+63927',
 '+63927',
 '+63917',
 '+63927',
 '+63917',
 '+63917',
 '+63917',
 '+63927',
 '+63917',
 '+63917',
 '+63918',
 '+63918',
 '+63927',
 '+63917',
 '+63927',
 '+63927',
 '+63927',
 '+63927',
 '+63918',
 '+63918',
 '+63918',
 '+63918',
 '+63918',
 '+63917',
 '+63917',
 '+63917',
 '+63917',
 '+63917',

In [165]:
len(beginning_phonenumbers_list) # to check if all numbers were accounted for

3000

In [166]:
mytally_network = {}

for word in beginning_phonenumbers_list:
    mytally_network.setdefault(word,beginning_phonenumbers_list.count(word))
    
mytally_network # not clean

{'+63918': 955, '+63927': 1019, '+63917': 1026}

In [172]:
globe_tally = mytally_network["+63927"] + mytally_network["+63917"]
globe_tally # combine the 2 globes

2045

In [174]:
smart_tally = mytally_network["+63918"]
smart_tally # clean referencing for the smart tally

955

In [175]:
mytally_network_clean = {"Globe": globe_tally, "Smart": smart_tally}
mytally_network_clean # make it clean

{'Globe': 2045, 'Smart': 955}

In [182]:
from IPython.core.display import HTML

html = """
<html>
    <body>
         <h2>Summary Statistics</h2>
         <table>
             <tr><th>Stat</th><th>Value</th></tr>
             <tr><td>People born in each month</td><td>{birthmonth_summary}</td></tr>
             <tr><td>People born on each day</td><td>{birthday_summary}</td></tr>
             <tr><td>People born in each year</td><td>{birthyear_summary}</td></tr>
             <tr><td>People coming from each course</td><td>{course_summary}</td></tr>
             <tr><td>People having numbers from Globe and Smart</td><td>{network_summary}</td></tr>
         </table>
    </body>
</html>
"""

html_output = html.format(birthmonth_summary = mytally_birthmonths,
                    birthday_summary = mytally_birthdays,
                    birthyear_summary = mytally_birthyears_sorted,
                    course_summary = mytally_courses,
                    network_summary = mytally_network_clean)

HTML(html_output)

In [181]:
from IPython.core.display import HTML

html = """
<html>
    <body>
         <h2>Summary Statistics</h2>
         <table>
             <tr><th>Stat</th><th>Value</th></tr>
             <tr><td>People born in each month</td><td>{birthmonth_summary}</td></tr>
             <tr><td>People born on each day</td><td>{birthday_summary}</td></tr>
             <tr><td>People born in each year</td><td>{birthyear_summary}</td></tr>
             <tr><td>People coming from each course</td><td>{course_summary}</td></tr>
             <tr><td>People having numbers from Globe and Smart</td><td>{network_summary}</td></tr>
         </table>
    </body>
</html>
"""

html_output = html.format(birthmonth_summary = mytally_birthmonths,
                    birthday_summary = mytally_birthdays,
                    birthyear_summary = mytally_birthyears_sorted,
                    course_summary = mytally_courses,
                    network_summary = mytally_network_clean)

HTML(html_output)

with open('basicstats.html',"w+") as file:
    file.write(html)

In [180]:
# open using Safari browser

html2 = """
    <html>
        <head>
            <link rel="stylesheet" type="css" href="Paris.css" />
        </head>
        <body>
            <div id="demoFont">Summary Statistics</div>
            <h2></h2>
            <div id="Font2">birthmonth tally</div>
            <div class="divTable blueTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Birthmonth</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            <div class="divTableRow">
            <div class="divTableCell">January</div><div class="divTableCell">270</div></div>
            <div class="divTableRow">
            <div class="divTableCell">February</div><div class="divTableCell">232</div></div>
            <div class="divTableRow">
            <div class="divTableCell">March</div><div class="divTableCell">246</div></div>
            <div class="divTableRow">
            <div class="divTableCell">April</div><div class="divTableCell">232</div></div>
            <div class="divTableRow">
            <div class="divTableCell">May</div><div class="divTableCell">273</div></div>
            <div class="divTableRow">
            <div class="divTableCell">June</div><div class="divTableCell">259</div></div>
            <div class="divTableRow">
            <div class="divTableCell">July</div><div class="divTableCell">249</div></div>
            <div class="divTableRow">
            <div class="divTableCell">August</div><div class="divTableCell">241</div></div>
            <div class="divTableRow">
            <div class="divTableCell">September</div><div class="divTableCell">232</div></div>
            <div class="divTableRow">
            <div class="divTableCell">October</div><div class="divTableCell">178</div></div>
            <div class="divTableRow">
            <div class="divTableCell">November</div><div class="divTableCell">277</div></div>
            <div class="divTableRow">
            <div class="divTableCell">December</div><div class="divTableCell">276</div></div>
            </div>
            <h2></h2>
            <div id="Font2">birthday tally</div>
            <div class="divTable blueTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Day</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            <div class="divTableRow">
            <div class="divTableCell">1</div><div class="divTableCell">112</div></div>
            <div class="divTableRow">
            <div class="divTableCell">2</div><div class="divTableCell">106</div></div>
            <div class="divTableRow">
            <div class="divTableCell">3</div><div class="divTableCell">90</div></div>
            <div class="divTableRow">
            <div class="divTableCell">4</div><div class="divTableCell">110</div></div>
            <div class="divTableRow">
            <div class="divTableCell">5</div><div class="divTableCell">100</div></div>
            <div class="divTableRow">
            <div class="divTableCell">6</div><div class="divTableCell">99</div></div>
            <div class="divTableRow">
            <div class="divTableCell">7</div><div class="divTableCell">88</div></div>
            <div class="divTableRow">
            <div class="divTableCell">8</div><div class="divTableCell">111</div></div>
            <div class="divTableRow">
            <div class="divTableCell">9</div><div class="divTableCell">98</div></div>
            <div class="divTableRow">
            <div class="divTableCell">10</div><div class="divTableCell">111</div></div>
            <div class="divTableRow">
            <div class="divTableCell">11</div><div class="divTableCell">105</div></div>
            <div class="divTableRow">
            <div class="divTableCell">12</div><div class="divTableCell">98</div></div>
            <div class="divTableRow">
            <div class="divTableCell">13</div><div class="divTableCell">119</div></div>
            <div class="divTableRow">
            <div class="divTableCell">14</div><div class="divTableCell">102</div></div>
            <div class="divTableRow">
            <div class="divTableCell">15</div><div class="divTableCell">121</div></div>
            <div class="divTableRow">
            <div class="divTableCell">16</div><div class="divTableCell">115</div></div>
            <div class="divTableRow">
            <div class="divTableCell">17</div><div class="divTableCell">116</div></div>
            <div class="divTableRow">
            <div class="divTableCell">18</div><div class="divTableCell">107</div></div>
            <div class="divTableRow">
            <div class="divTableCell">19</div><div class="divTableCell">113</div></div>
            <div class="divTableRow">
            <div class="divTableCell">20</div><div class="divTableCell">91</div></div>
            <div class="divTableRow">
            <div class="divTableCell">21</div><div class="divTableCell">123</div></div>
            <div class="divTableRow">
            <div class="divTableCell">22</div><div class="divTableCell">103</div></div>
            <div class="divTableRow">
            <div class="divTableCell">23</div><div class="divTableCell">116</div></div>
            <div class="divTableRow">
            <div class="divTableCell">24</div><div class="divTableCell">108</div></div>
            <div class="divTableRow">
            <div class="divTableCell">25</div><div class="divTableCell">116</div></div>
            <div class="divTableRow">
            <div class="divTableCell">26</div><div class="divTableCell">116</div></div>
            <div class="divTableRow">
            <div class="divTableCell">27</div><div class="divTableCell">102</div></div>
            <div class="divTableRow">
            <div class="divTableCell">28</div><div class="divTableCell">104</div></div>
            </div>
            <h2></h2>
            <div id="Font2">birthyear tally</div>
            <div class="divTable blueTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Birthyear</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            <div class="divTableRow">
            <div class="divTableCell">1997</div><div class="divTableCell">756</div></div>
            <div class="divTableRow">
            <div class="divTableCell">1998</div><div class="divTableCell">746</div></div>
            <div class="divTableRow">
            <div class="divTableCell">1999</div><div class="divTableCell">756</div></div>
            <div class="divTableRow">
            <div class="divTableCell">2000</div><div class="divTableCell">380</div></div>
            <div class="divTableRow">
            <div class="divTableCell">2001</div><div class="divTableCell">362</div></div>
            <h2></h2>
            <div id="Font2">course tally</div>
            <div class="divTable blueTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Course</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            <div class="divTableRow">
            <div class="divTableCell">BS Legal Management</div><div class="divTableCell">585</div></div>
            <div class="divTableRow">
            <div class="divTableCell">BS Management</div><div class="divTableCell">629</div></div>
            <div class="divTableRow">
            <div class="divTableCell">BS Communications Technology</div><div class="divTableCell">589</div></div>
            <div class="divTableRow">
            <div class="divTableCell">BS Information Technology Entrepreneurship</div><div class="divTableCell">590</div></div>
            <div class="divTableRow">
            <div class="divTableCell">BS Management Engineering</div><div class="divTableCell">607</div></div>
            <h2></h2>
            <div id="Font2">network tally</div>
            <div class="divTable blueTable">
            <div class="divTableHeading">
            <div class="divTableRow">
            <div class="divTableHead">Network</div>
            <div class="divTableHead">Tally</div>
            </div>
            </div>
            <div class="divTableBody">
            <div class="divTableRow">
            <div class="divTableCell">Globe</div><div class="divTableCell">2045</div></div>
            <div class="divTableRow">
            <div class="divTableCell">Smart</div><div class="divTableCell">955</div></div>
            </div>
        </body>
    </html>
"""

with open('stats.html',"w+") as file:
    file.write(html2)